events/worker.go

```diff
@@ -1,40 +1,44 @@
 package events
 
 import (
-	"encoding/json"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/rancher/event-subscriber/locks"
 	"github.com/rancher/go-rancher/client"
 )
 
-func newWorker() *Worker {
-	return &Worker{}
-}
+type ChanWorkerPool chan int
 
-type Worker struct {
+func WorkerPool(size int) ChanWorkerPool {
+	wp := ChanWorkerPool(make(chan int, size))
+	for i := 0; i < size; i++ {
+		wp <- i
+	}
+	return wp
 }
 
-func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler, apiClient *client.RancherClient,
-	workers chan *Worker) {
-	defer func() { workers <- w }()
-
-	event := &Event{}
-	err := json.Unmarshal(rawEvent, &event)
-	if err != nil {
+func (wp ChanWorkerPool) HandleWork(message *Event, eventHandlers map[string]EventHandler, apiClient *client.RancherClient, locker locks.Locker) {
+	select {
+	case w := <-wp:
+		go func() {
+			defer func() { wp <- w }()
+			doWork(message, eventHandlers, apiClient, locker)
+		}()
+	default:
 		log.WithFields(log.Fields{
-			"err": err,
-		}).Error("Error unmarshalling event")
-		return
+			"workerCount": cap(wp),
+		}).Info("No workers available dropping event.")
```

Comment:
I think this is important and abnormal enough to be logged at warn level. Can you make it warn?